docs(readme): Add missing examples (#113)
stackbuilders/assertive-ts: README.md

<!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->

[![All Contributors](https://img.shields.io/badge/all_contributors-10-orange.svg?style=flat-square)](#contributors-)

<!-- ALL-CONTRIBUTORS-BADGE:END -->

> 

Reviewer: Let's remove these extra spaces. That section is managed by all-contributors, we should never change it

This library provides **fluent assertions**, which means you can chain multiple matcher functions to the same value under test:

```ts

> expect("assertive-ts is awesome!").toStartWith("assertive-ts").not.toContain("unsafe").toEndWith("awesome!");

Reviewer: Let's revert this change (maybe caused by an IDE formatter?). The code was intentionally stacked for better readability.
```suggestion
expect("assertive-ts is awesome!")
  .toStartWith("assertive-ts")
  .not.toContain("unsafe")
  .toEndWith("awesome!");
```

// Number assertion
expect(sum(1, 2)).toBePositive();

// Error assertion
expect(new Error(errorMessage)).toHaveMessage(expectedError);

// Array assertion
const data = [1, 2, 3, 4]
expect(data).toMatchAll(x => x < 5);
expect(data).toBeEmpty()

// Date assertion
const date = new Date(2023, 12, 31);
expect(date).toBeAfter(new Date(2023, 12, 1));
expect(date).toBeBefore(new Date(2024, 1, 1));

// Object assertion
const objectData = {
  key1: "test1",
  key2: "test2",
};
expect(objectData).toContainKey("key1");
expect(objectData).toContainEntry(["key1", "test1"]);

expect(14).toEndWith("4");
           ^ ? type error: `toEndWith` does not exist in `NumberAssertion`
```

> You can also assert from synchronous and asynchronous code, for example:

Reviewer: ```suggestion
You can also assert over functions and asynchronous code, for example:
```

```ts

> const verifyEnvVar = (): void => {
>   const { MY_ENV_VAR } = process.env;
> 
>   if (!MY_ENV_VAR) {
>     throw new Error("Missing MY_ENV_VAR environment variable");
>   }
> };

Reviewer: Top-level functions are usually better as named function declarations instead of arrow function expressions because of [hoisting](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting) of the declaration and the scope of the `this` value. Here's a great 3mins read that dives deep into this: https://kentcdodds.com/blog/function-forms

In this case, were not running this code, but we should still keep good practices and conventions 🙂 
```suggestion
function verifyEnvVar(): void {
  const { MY_ENV_VAR } = process.env;

  if (!MY_ENV_VAR) {
    throw new Error("Missing MY_ENV_VAR environment variable");
  }
}
```

// assertion
expect(() => verifyEnvVar())
  .toThrowError(Error)
  .toHaveMessage("Missing MY_ENV_VAR environment variable");

expect(() => verifyEnvVar()).not.toThrow();

> export const getData = async (): Promise<DataType> => {
>   const data = await requestApi();
> 
>   if (!data) {
>     throw new Error("Data was not found");
>   }
> 
>   return data;
> };

Reviewer: Same here. Also, I think we can remove the `export` 🙂 
```suggestion
async function getData(): Promise<DataType> {
  const data = await requestApi();

  if (!data) {
    throw new Error("Data was not found");
  }

  return data;
}
```
